### DISCUSSION TITLE
blog/dockerizing-a-nuxt-app

### DISCUSSION THREAD

**Discussion by giscus** [Mokkapps/mokkapps-de-discussions#13] [Announcements]
# blog/dockerizing-a-nuxt-app

Learn how to dockerize your Nuxt 3+ application to ensure consistency, isolation, scalability, and portability across different environments. This guide walks you through the steps to create a Docker container for your Nuxt 3+ app, making deployment seamless and efficient.

https://mokkapps.de/blog/dockerizing-a-nuxt-app?source=weeklyVueNews&campaign=154

**Comment by ralacerda** [comment]
Amazing post! I was just trying to do that a few days ago.
But it left me wondering two things:
- Are the big differences between the `slim` and the `alpine` version?
- Will the container have the source files and the original `node_modules`? For a case where I need the smallest image possible, how can I keep only the `.output` folder?

**Reply by Mokkapps** [reply]
> Amazing post! I was just trying to do that a few days ago. But it left me wondering two things:
> 
> * Are the big differences between the `slim` and the `alpine` version?

I suppose the `slim` version is less big and has only the most important dependencies installes. But I never did a full comparison between both. 

> * Will the container have the source files and the original `node_modules`? For a case where I need the smallest image possible, how can I keep only the `.output` folder?

No, in line 34 we only copy the `.output` folder to the working directory: `COPY --from=build /app/.output ./`

**Comment by Zadigo** [comment]
I tried the code on a brad new Nuxt project without any extra dependencies and I am getting this error:

```javascript
app_forum_dev  | node:internal/modules/cjs/loader:1228
app_forum_dev  |   throw err;
app_forum_dev  |   ^                                                                                  
app_forum_dev  |                                                                                      
app_forum_dev  | Error: Cannot find module '/app/pnpm'
app_forum_dev  |     at Function._resolveFilename (node:internal/modules/cjs/loader:1225:15)
app_forum_dev  |     at Function._load (node:internal/modules/cjs/loader:1055:27)
app_forum_dev  |     at TracingChannel.traceSync (node:diagnostics_channel:322:14)
app_forum_dev  |     at wrapModuleLoad (node:internal/modules/cjs/loader:220:24)
app_forum_dev  |     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:170:5)                                                                                                  
app_forum_dev  |     at node:internal/main/run_main_module:36:49 {
app_forum_dev  |   code: 'MODULE_NOT_FOUND',
app_forum_dev  |   requireStack: []
app_forum_dev  | }
app_forum_dev  | 
app_forum_dev  | Node.js v22.14.0
app_forum_dev exited with code 1
```

**Reply by Zadigo** [reply]
No worries, it was a problem on my side ^^
